tests/conftest.py

```diff
@@ -1675,54 +1675,65 @@ def tmpvenv_active_inherit(deactive_venv, tmpdir: py.path.local) -> Iterator[env
 
 
 @pytest.fixture(scope="session")
-def local_module_package_index(modules_v2_dir: str) -> Iterator[str]:
+def create_local_package_index():
     """
-    Creates a local pip index for all v2 modules in the modules v2 dir. The modules are built and published to the index.
-    :return: The path to the index
+    A fixture that provides a function to create a local pip package index
+    from all v2 modules in a specified directory. A name can also be given to the index
     """
-    cache_dir = os.path.abspath(os.path.join(os.path.dirname(modules_v2_dir), f"{os.path.basename(modules_v2_dir)}.cache"))
-    build_dir = os.path.join(cache_dir, "build")
-    index_dir = os.path.join(build_dir, "simple")
-    timestamp_file = os.path.join(cache_dir, "cache_creation_timestamp")
-
-    def _should_rebuild_cache() -> bool:
-        if any(not os.path.exists(f) for f in [build_dir, index_dir, timestamp_file]):
-            # Cache doesn't exist
-            return True
-        if len(os.listdir(index_dir)) != len(os.listdir(modules_v2_dir)) + 3:  # #modules + index.html + setuptools + wheel
-            # Modules were added/removed from the build_dir
-            return True
-        # Cache is dirty
-        return any(
-            os.path.getmtime(os.path.join(root, f)) > os.path.getmtime(timestamp_file)
-            for root, _, files in os.walk(modules_v2_dir)
-            for f in files
-            if "egg-info" not in root  # we write egg info in some test, messing up the tests
-        )
 
-    if _should_rebuild_cache():
-        logger.info(f"Cache {cache_dir} is dirty. Rebuilding cache.")
-        # Remove cache
-        if os.path.exists(cache_dir):
-            shutil.rmtree(cache_dir)
-        os.makedirs(build_dir)
-        # Build modules
-        for module_dir in os.listdir(modules_v2_dir):
-            path: str = os.path.join(modules_v2_dir, module_dir)
-            ModuleTool().build(path=path, output_dir=build_dir)
-        # Download bare necessities
-        CommandRunner(logging.getLogger(__name__)).run_command_and_log_output(
-            ["pip", "download", "setuptools", "wheel"], cwd=build_dir
-        )
+    def _create_local_package_index(modules_v2_dir: str, index_name: str = "simple"):
```

Comment:
This method uses a cache directory that is created at the same level in the filesystem as the `modules_v2_dir`. This cache is useful for the index that is built for the `modules_v2_dir`, because like that we don't need to build the index over and over again when re-running the test suite. But this cache is not useful for the `test_process_env_install_from_index_not_found_env_var` test case. That test case passes in a `tmpdir` as `modules_v2_dir`. This means that the cache directory is created outside of the tmpdir. This cache directory is never re-used and never cleaned up.

That said, I think you are making things way too complicated for yourself. All you need is a fixture that returns an empty pip index. So it would be sufficient to create a new fixture that executes `dir2pi(argv=["dir2pi", build_dir])` for an empty `build_dir` and yield the index directory `os.path.join(build_dir, "simple")`. Like that you can keep the `local_module_package_index` fixture as it was.